### PR TITLE
Add +x variant to petsc.

### PR DESCRIPTION
**Pull request by s-sajid-ali**

### CODE REVIEW

**Comment by pramodk** [comment]
does this need then dependency with x11?

**Comment by s-sajid-ali** [comment]
Should I make it +X instead of +x to maintain consistency with [this](https://github.com/spack/spack/pull/9722) ?

**Comment by davydden** [comment]
> Should I make it +X instead of +x to maintain consistency with this ?
 
I am away-from-keyboard, but one could grep other packages to see what do we use in Spack more often.

File: var/spack/repos/builtin/packages/petsc/package.py [hunk 1]
```diff
@@ -210,6 +213,11 @@ def install(self, spec, prefix):
             '--with-blas-lapack-lib=%s' % lapack_blas.joined()
         ])
 
+        if 'x' in spec:
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/petsc/package.py]: I'm not 100% sure that this works. I would use this instead:
```suggestion
        if '+x' in spec:
```

**Pull request merged**